File: app/src/main/java/com/android3/siegertpclient/data/setting/settingsource/settingLocal/SettingLocalDataSource.java
package com.android3.siegertpclient.data.setting.settingsource.settingLocal;

import com.android3.siegertpclient.data.setting.Setting;

import org.jetbrains.annotations.NotNull;


public class SettingLocalDataSource {

    final SettingDao settingDao;

    public SettingLocalDataSource(SettingDao settingDao) {
        this.settingDao = settingDao;
    }

    public boolean check(@NotNull String Language) {

        if(Language.equals("English"))  {
            return true;
        }
        if(Language.equals("German")){
            return true;
        }
        return false;
    }

    public void saveSetting(@NotNull Setting newSetting) {
        settingDao.insertSetting(newSetting);
    }

}

File: app/src/main/java/com/android3/siegertpclient/data/setting/settingsource/settingLocal/SettingLocalDataSource.kt
package com.android3.siegertpclient.data.setting.settingsource.settingLocal

import com.android3.siegertpclient.data.Result.Result
import com.android3.siegertpclient.data.setting.Setting

/**
 * class setting local data source is used to set the local data source.
 */
class SettingLocalDataSource(val settingDao: SettingDao) {

    fun saveSetting(newSetting: Setting) {
        settingDao.insertSetting(newSetting)
    }
    fun getSetting(){
        settingDao.allSettings
    }
    fun deleteSetting(){
        settingDao.deleteAllSetting()
    }
    fun updateSetting(setting:Setting){
        settingDao.updateSetting(setting)
    }
}